### PR TITLE
[Collections] Removed local package collection comes back alive

### PR DESCRIPTION
**Pull request by yim-lee**
Steps to reproduce:
1. Add a local collection
2. Delete the collection **file** on local disk
3. Refresh collections
4. The unavailable local collection is back!

This is caused by us not deleting unavailable/invalid collections from storage when refresh fails.

rdar://80388968

### CODE REVIEW

**Comment by yim-lee** [comment]
@swift-ci please smoke test

File: Sources/PackageCollections/PackageCollections.swift [hunk 1]
```diff
@@ -454,16 +454,16 @@ public struct PackageCollections: PackageCollectionsProtocol {
     private func refreshCollectionFromSource(source: PackageCollectionsModel.CollectionSource,
                                              trustConfirmationProvider: ((PackageCollectionsModel.Collection, @escaping (Bool) -> Void) -> Void)?,
                                              callback: @escaping (Result<Model.Collection, Error>) -> Void) {
-        if let errors = source.validate()?.errors() {
```
Comment (yim-lee) [inline Sources/PackageCollections/PackageCollections.swift:457]: We have the same logic in the provider. Removing this so we don't have to duplicate storage removal logic (i.e., L464-466) here.

**Comment by tomerd** [comment]
what is the expected behavior if the collection is on the network and there is a network issue? i.e. the refresh fails because it cant reach the server.... would deleting the collection still make sense?

**Comment by yim-lee** [comment]
The collection URL remains in the config file so on next refresh it will become available again if the network issue goes away or the local file is placed back.

**Review by tomerd — APPROVED** [review]

**Comment by yim-lee** [comment]
We can distinguish between a) bad local file, b) not found (404), and c) unavailable (non-404 error codes) such that we only remove from storage in cases (a) and (b). Do we want to do that?

**Comment by tomerd** [comment]
> We can distinguish between a) bad local file, b) not found (404), and c) unavailable (non-404 error codes) such that we only remove from storage in cases (a) and (b). Do we want to do that?

> But then, we can't tell in case of (c) if it's temporary or permanent.

if the next refresh will try to fetch it I think its fine

**Comment by yim-lee** [comment]
```
21:17:38 Test Case 'BuildPlanTests.testExplicitSwiftPackageBuild' started at 2021-07-15 21:15:43.762
21:17:38 SwiftDriver/ClangVersionedDependencyResolution.swift:165: Fatal error: Unresolved placeholder dependencies at planning stage: swiftPlaceholder("C")
21:17:38 Exited with signal code 4
```

@swift-ci please smoke test Linux

**Comment by tomerd** [comment]
the linux CI failure seems like a regression in driver, other PRs are failing the same way cc @artemcm

**Comment by tomerd** [comment]
@swift-ci please smoke test linux

**Pull request merged**